tests/bor/bor_reorg_test.go

```diff
@@ -276,3 +144,136 @@ func TestValidatorWentOffline(t *testing.T) {
 	assert.Equal(t, authorVal1, nodes[0].AccountManager().Accounts()[0])
 
 }
+
+func TestForkWithBlockTime(t *testing.T) {
+
+	cases := []struct {
+		sprint        uint64
+		blockTime     map[string]uint64
+		change        uint64
+		producerDelay uint64
+		forkExpected  bool
+	}{
+		{
+			sprint: 128,
+			blockTime: map[string]uint64{
+				"0":   5,
+				"128": 2,
+				"256": 8,
+			},
+			change:        2,
+			producerDelay: 8,
+			forkExpected:  false,
+		},
+		{
+			sprint: 64,
+			blockTime: map[string]uint64{
+				"0":  5,
+				"64": 2,
+			},
+			change:        1,
+			producerDelay: 5,
+			forkExpected:  false,
+		},
+		{
+			sprint: 16,
+			blockTime: map[string]uint64{
+				"0":  2,
+				"64": 5,
+			},
+			change:        4,
+			producerDelay: 4,
+			forkExpected:  true,
+		},
+	}
+
+	// Create an Ethash network based off of the Ropsten config
+	genesis := initGenesis(t)
+
+	for _, test := range cases {
+		genesis.Config.Bor.Sprint = test.sprint
```

Comment:
can we make it as subtests?

Reply:
[3e4b872](https://github.com/maticnetwork/bor/pull/535/commits/3e4b872e16081c057518617d9ee803c372ec33a2)